### PR TITLE
api interop layer deploy

### PR DESCRIPTION
**Pull request by jamestranovich-noaa**
## What does this PR do? 🛠️

Addresses https://github.com/weather-gov/weather.gov/issues/1791 -- except, in james, and not sandbox, since I don't think we're quite set up on that front (yet)

## What does the reviewer need to know? 🤔

This only works in `james` currently. Visit https://api-weathergov-james.app.cloud.gov/ (by default the app will return `{ ok: true }` for `/`)

Also, https://weathergov-james.app.cloud.gov/ uses the interop layer by default now. Feel free to poke around.

Also, NR logs are enabled: 

image

### CODE REVIEW

**Review by eric-gade — APPROVED** [review]
LGTM
  
One general question though -- why is the main app only set to `256M`? Seems small?

**Comment by jamestranovich-noaa** [comment]
@greg-does-weather @eric-gade i've made some changes to this PR, mostly making sure all of the env variables are properly propagated. please take another look :eyes:

**Comment by jamestranovich-noaa** [comment]
> One general question though -- why is the main app only set to `256M`? Seems small?

Yes. cloud.gov apps don't need that much memory because (ostensibly) it's running only that one app. The good thing is that NR will track app utilization for us and we can set warnings if we are running out of memory or disk or whatever. 

It looks like we currently have 5G across _all_ environments/instances. (To check the quota, you can get the quota id from `cf org nws-weathergov` and then doing `cf quota <id>`)

**Comment by eric-gade** [comment]
Note that this also closes the specific issue outlined in #1786

File: web/modules/weather_data/src/Service/DataLayer.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function __construct(
     private function fetch($url, $attempt = 1, $delay = 75)
     {
         if (!preg_match("/^https?:\/\//", $url)) {
-            $baseUrl = getEnv("API_URL");
+            $baseUrl = getEnv("API_PROXY_URL");
```
Comment (greg-does-weather) [inline web/modules/weather_data/src/Service/DataLayer.php:69]:
```suggestion
            $baseUrl = getEnv("API_URL");
```

File: api-interop-layer/util/fetch.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { sleep } from "./sleep.js";
 
 const logger = createLogger("fetch wrapper");
 
-const BASE_URL = process.env.API_URL ?? "https://api.weather.gov";
+const BASE_URL = process.env.API_PROXY_URL ?? "https://api.weather.gov";
```
Comment (greg-does-weather) [inline api-interop-layer/util/fetch.js:6]:
```suggestion
const BASE_URL = process.env.API_URL ?? "https://api.weather.gov";
```

File: docker-compose.test.yml [hunk 3]
```diff
@@ -10,8 +10,8 @@ services:
       - api-proxy-test
       - api-interop-layer-test
     environment: # we want to extend, not override
-      API_URL: http://api-proxy-test:9081
-      INTEROP_URL: http://api-interop-layer-test:9082
+      API_PROXY_URL: http://api-proxy-test:9081
```
Comment (greg-does-weather) [inline docker-compose.test.yml:13]:
```suggestion
      API_URL: http://api-proxy-test:9081
```

File: docker-compose.yml [hunk 4]
```diff
@@ -36,8 +36,8 @@ services:
       DB_PASSWORD: drupal
       DB_HOST: database
       DB_PORT: 3306
-      API_URL: http://api-proxy:8081
-      APP_NAME: ${APP_NAME:-local}
+      API_PROXY_URL: http://api-proxy:8081
```
Comment (greg-does-weather) [inline docker-compose.yml:39]:
```suggestion
      API_URL: http://api-proxy:8081
```

File: docker-compose.yml [hunk 5]
```diff
@@ -67,8 +67,8 @@ services:
       - api-proxy
       - api-interop-layer
     environment:
-      API_URL: http://api-proxy:8081
-      INTEROP_URL: http://api-interop-layer:8082
+      API_PROXY_URL: http://api-proxy:8081
```
Comment (greg-does-weather) [inline docker-compose.yml:70]:
```suggestion
      API_URL: http://api-proxy:8081
```

File: manifests/manifest-design.yaml [hunk 6]
```diff
@@ -9,7 +9,7 @@ default_config: &defaults
   env:
     PHP_INI_SCAN_DIR: /home/vcap/app/php/etc/:/home/vcap/app/php/etc/php.ini.d/
     NEWRELIC_LICENSE: ((newrelic-license))
-    API_URL: https://weathergov-api-proxy.app.cloud.gov
+    API_PROXY_URL: https://weathergov-api-proxy.app.cloud.gov
```
Comment (greg-does-weather) [inline manifests/manifest-design.yaml:12]:
```suggestion
    API_URL: https://weathergov-api-proxy.app.cloud.gov
```

**Review by greg-does-weather — APPROVED** [review]

**Pull request merged**